### PR TITLE
Fix return void when multiple return types specified

### PR DESCRIPTION
**Pull request by deizel**
Happy to rebase if you want to merge #134 first.

### CODE REVIEW

**Comment by bcrowe** [comment]
@deizel I merged the other PR.

**Comment by deizel** [comment]
@bcrowe Thanks, rebased.

**Comment by bcrowe** [comment]
Looks like this causes methods with a `@return mixed`-only annotation and contains a `return;` statement to emit warnings.

**Comment by deizel** [comment]
There is a test for that but it passes unless I add a description after `@return mixed`. I'll take a closer look.

**Comment by deizel** [comment]
Ran it against core this time and everything seems to be passing as expected.

File: CakePHP/tests/files/function_comment_opening_line_fail.php [hunk 1]
```diff
@@ -1,13 +1,14 @@
 <?php
+namespace Beakman;
 
-class Foo {
+class Foo
+{
     /** 
-     * There is a (disallowed) trailing space after the opening ** in this doc block.
+     * ^ There is a (disallowed) trailing space after the opening ** in this doc block.
```
Comment (markstory) [inline CakePHP/tests/files/function_comment_opening_line_fail.php]: Why is this here?

Comment (deizel) [inline CakePHP/tests/files/function_comment_opening_line_fail.php]: To help point out the (disallowed) trailing space above it. :stuck_out_tongue_winking_eye:

Comment (markstory) [inline CakePHP/tests/files/function_comment_opening_line_fail.php]: Oh derp.

File: CakePHP/Sniffs/Commenting/FunctionCommentSniff.php [hunk 2]
```diff
@@ -115,89 +115,94 @@ protected function processReturn(PHP_CodeSniffer_File $phpcsFile, $stackPtr, $co
...
+        $content = $tokens[($return + 2)]['content'];
+        if (empty($content) === true || $tokens[($return + 2)]['code'] !== T_DOC_COMMENT_STRING) {
+            $error = 'Return type missing for @return tag in function comment';
+            $phpcsFile->addError($error, $return, 'MissingReturnType');
+            return;
+        }
+
+        // Check return type (can be multiple, separated by '|').
+        list($types,)   = explode(' ', $content);
+        $typeNames      = explode('|', $types);
```
Comment (bcrowe) [inline CakePHP/Sniffs/Commenting/FunctionCommentSniff.php]: Nitpick: I don't think we typically whitespace-align assignments. Other than this, everything LGTM. Nice work :clap:.

Comment (deizel) [inline CakePHP/Sniffs/Commenting/FunctionCommentSniff.php]: Neither do I :) Seems this class in particular had a lot of that, I've tidied it up in the next commit.

**Comment by deizel** [comment]
:+1:
